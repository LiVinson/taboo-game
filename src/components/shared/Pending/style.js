import styled from "styled-components"


export const StyledPending = styled.p`
  font-size: 1.5rem;
  font-style: italic;
  color: ${(props) => props.theme.color.primary}
  `

export const StyledLargePending = styled.p`
padding: 3.5rem .5rem;
font-size: 3.5rem;
color: ${(props) => props.theme.color.grayDark2}
`