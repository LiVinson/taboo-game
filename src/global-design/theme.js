const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
