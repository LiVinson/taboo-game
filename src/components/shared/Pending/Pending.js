import React from 'react'
import PropTypes from 'prop-types'
import { StyledPending, StyledLargePending } from './style'

export class Pending extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			message: this.props.message,
		}
	}

	componentDidMount() {
		const { speed, message } = this.props

		this.setPendingMessage(message, speed)
	}

	setPendingMessage = (message, speed) => {
		this.interval = window.setInterval(() => {
			this.state.message === message + '...'
				? this.setState({ message: message })
				: this.setState(({ message }) => ({ message: message + '.' }))
		}, speed)
	}
	componentDidUpdate(prevProps) {
		if (prevProps.message !== this.props.message) {
			window.clearInterval(this.interval)
			this.setState(
				{
					message: this.props.message,
				},
				this.setPendingMessage(this.props.message, this.props.speed)
			)
		}
	}
	componentWillUnmount() {
		window.clearInterval(this.interval)
	}

	render() {
		const { large } = this.props
		const pending = large ? (
			<StyledLargePending>{this.state.message}</StyledLargePending>
		) : (
			<StyledPending>{this.state.message}</StyledPending>
		)
		return pending
	}
}

Pending.defaultProps = {
	speed: 300,
	large: false,
}

Pending.propTypes = {
	message: PropTypes.string.isRequired,
	speed: PropTypes.number.isRequired,
	large: PropTypes.bool.isRequired,
}
