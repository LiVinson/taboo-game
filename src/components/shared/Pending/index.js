import {Pending} from "./Pending"

export default Pending