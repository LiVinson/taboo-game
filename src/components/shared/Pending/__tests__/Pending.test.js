import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../../global-design/theme'
import { Pending } from '../Pending'

describe('Pending', () => {
	const defaultProps = {
		message: 'Calculating scores',
	}

	test('Renders default Pending message correctly', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<Pending {...props} />
				</ThemeProvider>
			)
			.toJSON()
		expect(wrapper).toMatchSnapshot()
	})

	test('Renders large Pending message correctly', () => {
		const props = {
			...defaultProps,
			large: true,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<Pending {...props} />
				</ThemeProvider>
			)
			.toJSON()
		expect(wrapper).toMatchSnapshot()
	})

	test('Adds elipses to message based on interval', () => {
		const props = {
			...defaultProps,
			speed: 500,
		}

		jest.useFakeTimers()
		const wrapper = mount(
			<ThemeProvider theme={theme}>
				<Pending {...props} />
			</ThemeProvider>
		)
		const component = wrapper.find(Pending)
		expect(component.text()).toBe(props.message)
		jest.advanceTimersByTime(1500)
		expect(component.text()).toBe(props.message + '...')
	})
})
