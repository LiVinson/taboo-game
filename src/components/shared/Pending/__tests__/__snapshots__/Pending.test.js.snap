// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pending Renders default Pending message correctly 1`] = `
.c0 {
  font-size: 1.5rem;
  font-style: italic;
  color: #1982C4;
}

<p
  className="c0"
>
  Calculating scores
</p>
`;

exports[`Pending Renders large Pending message correctly 1`] = `
.c0 {
  padding: 3.5rem .5rem;
  font-size: 3.5rem;
  color: #333333;
}

<p
  className="c0"
>
  Calculating scores
</p>
`;
